antarest/launcher/service.py

```diff
@@ -736,19 +736,36 @@ def get_load(self, from_cluster: bool = False) -> Dict[str, float]:
             load["local"] = float(local_used_cpus) / (os.cpu_count() or 1)
         return load
 
-    def get_versions(self, params: RequestParameters) -> Dict[str, List[str]]:
-        version_dict = {}
-        if self.config.launcher.local:
-            version_dict["local"] = list(
-                self.config.launcher.local.binaries.keys()
-            )
-
-        if self.config.launcher.slurm:
-            version_dict[
-                "slurm"
-            ] = self.config.launcher.slurm.antares_versions_on_remote_server
-
-        return version_dict
+    def get_solver_versions(self, solver: str) -> List[str]:
+        """
+        Fetch the list of solver versions from the configuration.
+
+        Args:
+            solver: name of the configuration to read: "default", "slurm" or "local".
+
+        Returns:
+            The list of solver versions.
+            This list is empty if the configuration is not available.
+
+        Raises:
+            ValueError: if the configuration is not "default", "slurm" or "local".
+        """
+        local_config = self.config.launcher.local
+        slurm_config = self.config.launcher.slurm
+        default_config = self.config.launcher.default
+        versions_map = {
+            "local": sorted(local_config.binaries) if local_config else [],
+            "slurm": sorted(slurm_config.antares_versions_on_remote_server)
+            if slurm_config
+            else [],
+        }
+        versions_map["default"] = versions_map[default_config]
+        try:
+            return versions_map[solver]
+        except KeyError:
+            raise ValueError(
+                f"Unknown solver configuration: '{solver}'"
+            ) from None
```

Comment:
we can use a helper function to reduce duplication and simplify:

```python
 def get_sorted_versions(solver_config, attribute):
            return sorted(getattr(solver_config, attribute)) if solver_config else []

        solvers = {
            "local": get_sorted_versions(self.config.launcher.local, "binaries"),
            "slurm": get_sorted_versions(self.config.launcher.slurm, "antares_versions_on_remote_server"),
        }

        solvers["default"] = solvers[self.config.launcher.default]

       return solvers.get(solver)
        ```

Reply:
Well, I prefer to keep the code simple.